### DISCUSSION TITLE
install mulit flux on one k8s in different namespace

### DISCUSSION THREAD

**Discussion by JingangLi** [fluxcd/flux2#3492] [Q&A] [answered]
Hi!

We want to use Flux, as it is begining, we have lots of team wants to use it for different work.

So we want to isolate their environement. Can we install multi flux on one kubernetes cluster's different namespace.
If we can limit the all installation with same flux version, is it supported case by flux?

And also we don't know the perfermence or footprint of flux, is there some suggest how much can be supported by one flux instance?

Many thanks.

Rgds,
Jingang Li

**Comment by makkes** [comment, marked as answer]
The Flux project doesn't generally recommend or support installing multiple Flux instances on the same cluster. For a guide on multi-tenant use cases you might find [this page](https://fluxcd.io/flux/security/best-practices/#additional-best-practices-for-shared-cluster-multi-tenancy) helpful.

**Reply by JingangLi** [reply]
Hi @makkes ,

Really thanks for your comments. It helps us a lot. 
Do you feel the flux will hit the perforemence issue with single instance on one cluster? It is for a reference for our work :-)

Rgds,
Jingang Li

**Reply by makkes** [reply]
In case you're hitting performance bottlenecks with Flux you can tweak it's configuration as documented [here](https://fluxcd.io/flux/cheatsheets/bootstrap/#increase-the-number-of-workers).

**Reply by JingangLi** [reply]
Hi @makkes ,

Many thaks for your help. We will use this when to tuning the flux deployment :-)

Have a nice day!

Rgds,
Jingang Li
